### PR TITLE
Add new ProductProvider and proof of concept.

### PR DESCRIPTION
**Pull request by floridamaniac**
`@Contexts/ProductIDContext` now has a few extra exports:

```
export {
  ProductIDContext,
  ProductProvider,
  useCurrentProduct,
  useCurrentProductId,
};
```

`ProductIDContext` still works as expected, so it won't break anything in anybody else's code. Ideally, we should probably switch over to using the custom `useCurrentProductId` hook and eventually deprecate `ProductIDContext`, but that's pretty low priority.

The biggest difference is that `useCurrentProductId` returns a destructurable object of `{ currentProductId, setCurrentProductId }`

### What does this mean?

This means that we can now CHANGE the Product ID from within our code, and it should trigger a fresh API call chain. Nobody has used this yet, but I believe @ryand8008 will be needing this for related products.

It also means we have a `useCurrentProduct` hook, which is similar but returns the Product object instead of just the ID.

Lastly, the providers have all been wrapped up and replaced in `index.jsx` with a `<ProductProvider>` jsx tag.

For testing purposes, you can keep changing the default product ID in `index.jsx` :)

This PR also includes a minor update to the Overview widget as a proof of concept that this is working.

### CODE REVIEW

**Review by ivyhu630 — APPROVED** [review]
Nice set up!! For people (aka me) who only need to access productID I still prefer to just access it directly  using  const productId = useContext(ProductIDContext) and not have to touch the state logic

**Comment by floridamaniac** [comment]
@ivyhu630 that should be fine! i think it could cause problems once Ryan implements clicking on a related product though. My understanding is that would trigger a re-render of a product so it might mean stuff is out of sync down the line. For MVP I think it's fine though.

**Comment by floridamaniac** [comment]
Either way, it's non-breaking so if that's something you want to switch to later the option is there, but you can keep using it as is too !

**Comment by ivyhu630** [comment]
> 

That's an interesting point. Thanks for explaining. I haven't thought much about the performance lag between useContext and useState. I'll make a note of this and test it out during optimization phase.

File: client/src/contexts/ProductIDContext.jsx [hunk 1]
```diff
@@ -1,4 +1,57 @@
...
+const CurrentProductId = React.createContext();
+const CurrentProduct = React.createContext();
+const ProductIDContext = React.createContext();
+
+function useCurrentProduct() {
+  const [currentProduct, setCurrentProduct] = React.useContext(CurrentProduct);
+  return { currentProduct, setCurrentProduct };
+}
+
+function useCurrentProductId() {
```
Comment (ryand8008) [inline client/src/contexts/ProductIDContext.jsx:15]: Very cool, will help with finding related products

**Review by ryand8008 — APPROVED** [review]
Thanks so much Walter!  I appreciate the ability to change things within the code to suit our needs at this junction.

**Pull request merged**